src/app/views/onboarding/step2.coffee

```diff
@@ -18,8 +18,24 @@ angular.module 'mnoEnterpriseAngular'
 
     # Select or deselect an app
     vm.toggleApp = (app) ->
-      # User cannot add more apps
-      return if vm.maxAppsSelected && !app.checked
+      # Add conflictingApp attribute to marketplace apps
+      app.subcategoryNames = []
+      _.each(app.subcategories, (appSubcategory) ->
+        app.subcategoryNames.push(appSubcategory.name)
+      )
+      app.marketplaceApps = _.filter(vm.marketplace.apps, (marketplaceApp) ->
+        marketplaceApp != app
+      )
+      _.each(app.marketplaceApps, (marketplaceApp) ->
+        _.each(marketplaceApp.subcategories, (marketplaceAppSubCategory) ->
+          _.each(app.subcategoryNames, (subcategoryName) ->
+            if !marketplaceAppSubCategory.multi_instantiable && (marketplaceAppSubCategory.name == subcategoryName)
+              marketplaceApp.conflictingApp = app
```

Comment:
I recommend that you read the lodash doc and play with it a bit, those functions are present in other languages as well :)

https://lodash.com/docs/3.10.1